Applications/Spire/Source/Ui/QuantitySpinBox.cpp

```diff
@@ -0,0 +1,65 @@
+#include "Spire/Ui/QuantitySpinBox.hpp"
+
+using namespace boost::signals2;
+using namespace Nexus;
+using namespace Spire;
+
+namespace {
+  auto real_to_quantity(RealSpinBox::Real value) {
+    auto quantity = Quantity::FromValue(value.str(
+      std::numeric_limits<Quantity>::digits10, std::ios_base::dec));
+    if(quantity != boost::none) {
```

Comment:
Feel free to start using "falsy" and if declarations:

    if(auto quantity = Quantity::FromValue(value.str(
        std::numeric_limits<Quantity>::digits10, std::ios_base::dec)) {
      return *quantity;
    }
    throw exception;